### ISSUE TITLE
WordPress4.9+TinyMCE Advanced+wysiwyg on Custom Post Type

### ISSUE THREAD

**Issue by shiget** [mgibbs189/custom-field-suite#413]
hi

When I upgraded WordPress to 4.9, the following problem occurred.

conditions:
  using TinyMCE Advanced plugin
  wysiswyg on custom post type

symptom:
  buttons of wysiwig are not displayed
  The following error is recorded on the console
     'Cannot read property 'indexOf' of undefined ...'

To temporarily avoid this problem, I made the following changes.
> plugins/custom-field-suite/includes/fields/wysiwyg.php

   Near line 130

                    // add the "code" button
                    if ( tinyMCE.settings.toolbar2.indexOf('code') < 0) {
                        tinyMCE.settings.toolbar2 += ',code';
                    }

                    to

                    // add the "code" button
                    //if ( tinyMCE.settings.toolbar2.indexOf('code') < 0) {
                        tinyMCE.settings.toolbar2 += ',code';
                    //}

I am fortunate if it is an issue that can be solved with the update.
thank you.

**Comment by holgerhubbs** [comment]
Wonderful! Hugs and kisses (-;

**Comment by kndt84** [comment]
I faced the same problem. Thx!

**Comment by austinpickett** [comment]
@shiget instead of commenting out that line I think it's more appropriate to do a type-check that tinyMCE.settings.toolbar2 isn't undefined. I've modified the line as shown:

`                    // add the "code" button
                    if ((typeof tinyMCE.settings.toolbar2 !== 'undefined') && (tinyMCE.settings.toolbar2.indexOf('code') < 0)) {
                        tinyMCE.settings.toolbar2 += ',code';
                    }`

**Comment by jcicero518** [comment]
I'm running into the same issue - toolbar2 is undefined so it throws an exception trying to get the indexOf property of it. I can't find any place where
 
`tinyMCE.settings.toolbar2` 

**would** be defined.

My alternative is to just wrap it in a try / catch block:

`// add the "code" button`
`try {`
`if ( tinyMCE.settings.toolbar2.indexOf('code') < 0) {`
`        tinyMCE.settings.toolbar2 += ',code';`
`    }`
`} catch (e) {`
 `   console.warn(e.message);`
` }`

This way it catches the exception and doesn't interfere with further execution but still gives a visual warning that something is wrong. Sorry about the formatting, I don't recall how I can add whitespace in MD.

**Comment by ulaulaman** [comment]
I have the same problems, so I performed some tests and found some problems that could be connected.
First of all: I created my original group of fields with CFS 2.5.9 and I update the plugin to CFS 2.5.11.
After the update from 4.8.+ to 4.9.+ the TinyMCE on custom post type disappears, but if I import fields in a new wordpress 4.9 installation, there's no problem, but it is impossible to modify the fields.
Furthermore I create a new group of fields and find that the TinyMCS disappears if I use e wysiwyg, so I think that this type of field could be the main problem.
This incompatibility also affect the publish box.
